### ISSUE TITLE
hash! type  member in objects is shared

### ISSUE THREAD

**Issue by Zamlox** [red/red#2343]
```
red>> o: object [block: make hash! 10]
== make object! [
    block: make hash! []
]
red>> a: make o []
== make object! [
    block: make hash! []
]
red>> b: make o []
== make object! [
    block: make hash! []
]
red>> append a/block [1 2 3 4]
== make hash! [1 2 3 4]
red>> a
== make object! [
    block: make hash! [1 2 3 4]
]
red>> b
== make object! [
    block: make hash! [1 2 3 4]
]
```
looks like ```a``` and ```b``` share same ```block```

**Comment by Zamlox** [comment]
Same issue for ```map!``` type

**Comment by dockimbel** [comment]
This is not a bug, `make` on object only copies by default some datatypes, for convenience, as COPYing is the most helpful behavior in the common case. So this applies to `any-string!`, `any-path!`, `block!` and `paren!`, but not for *heavier* datatypes like `hash!`, `map!` and `object!` which are shared by default.

**Comment by Zamlox** [comment]
All right, I was confused that it worked on ```block!``` datatype and I thought it should work on ```hash!``` and ```map!``` also.
Then perhaps I can either use ```block!``` datatype instead of ```hash!```, ```map!``` or I can take out the definition of that member from base object and move it to derived objects.
Sorry for confusion.

**Comment by qtxie** [comment]
Fixed in this commit:  be5663d
